python/lsst/consdb/cdb_schema.py

```diff
@@ -0,0 +1,317 @@
+                return member
+        return None
+
+
+########################
+# Schema preload class #
+########################
+
+
+class InstrumentTable:
+    """The column information for a single table in ConsDB schemas."""
+
+    def __init__(
+        self,
+        *,
+        engine: sqlalchemy.Engine,
+        get_db: Generator[Session, None, None],
+        instrument: str,
+        logger: logging.Logger,
+    ):
+        self.instrument = instrument.lower()
+        self.logger = logger
+        self.get_db = get_db
+
+        self.table_names = set()
+        self.schemas = dict()
+        self.flexible_metadata_schemas = dict()
+        self.obs_id_column = dict()
+        self.timestamp_columns = dict()
+
+        md = sqlalchemy.MetaData(schema=f"cdb_{self.instrument}")
+        md.reflect(engine)
+        self.table_names.update([str(table) for table in md.tables])
+        self.schemas = md
+        self.obs_id_column = dict()
+        self.flexible_metadata_schemas = dict()
+        for table in md.tables:
+            # Find all timestamp columns in the table
+            self.timestamp_columns[table] = set(
+                [
+                    column.name
+                    for column in md.tables[table].columns
+                    if isinstance(column.type, sqlalchemy.DateTime)
+                ]
+            )
+
+            for col_name in ObsIdColname:
+                col_name = col_name.value
+                if col_name in md.tables[table].columns:
+                    self.obs_id_column[table] = col_name
+                    break
+
+        for obs_type in ObsTypeEnum:
+            obs_type = obs_type.value
+            table_name = f"cdb_{self.instrument}.{obs_type}_flexdata"
+            schema_table_name = table_name + "_schema"
+            if table_name in md.tables and schema_table_name in md.tables:
+                schema_table = md.tables[schema_table_name]
+                stmt = sqlalchemy.select(schema_table.c["key", "dtype", "doc", "unit", "ucd"])
+                self.logger.debug(str(stmt))
+                schema = dict()
+                db = next(self.get_db())
+                for row in db.execute(stmt):
+                    schema[row[0]] = row[1:]
+                self.flexible_metadata_schemas[obs_type] = schema
+
+    def get_timestamp_columns(self, table: str) -> set[str]:
+        """Returns a set containing all timestamp columns.
+
+        Given the instrument and table name, returns a set
+        of strings listing all the columns in the table that
+        are a timestamp.
+
+        Parameters
+        ----------
+        table : `str`
+            The name of the table, e.g., "cdb_latiss.exposure".
+
+        Returns
+        -------
+        `set[str]`
+            The names of all timestamp columns in the table.
+        """
+        columns = self.timestamp_columns[table]
+        return columns
+
+    def get_schema_version(self) -> Version:
```

Comment:
I'm not sure how this might be used, but if we need it, we should pass it through from the source of truth rather than inferring it.  For a later ticket.